lib/std/zig/system.zig

```diff
@@ -199,14 +199,15 @@ pub const DetectError = error{
     OSVersionDetectionFail,
     Unexpected,
     ProcessNotFound,
-};
+} || mem.Allocator.Error;
 
 /// Given a `Target.Query`, which specifies in detail which parts of the
 /// target should be detected natively, which should be standard or default,
 /// and which are provided explicitly, this function resolves the native
 /// components by detecting the native system, and then resolves
 /// standard/default parts relative to that.
-pub fn resolveTargetQuery(query: Target.Query) DetectError!Target {
+/// The `arena` parameter is only used when detecting the native OS version.
+pub fn resolveTargetQuery(query: Target.Query, arena: mem.Allocator) DetectError!Target {
```

Comment:
I don't think we should make that commitment; better to just leave that unspecified and always pass an arena at call sites, I think. See: https://github.com/ziglang/zig/pull/24868#discussion_r2283117955